Documentation.md

```diff
@@ -164,590 +174,606 @@ and made available to your JS client code.
 that they can be routed incoming contacts.
 
 ### `agent.onNotRoutable()`
-```
-agent.onNotRoutable(function(agent) { ... });
+```js
+agent.onNotRoutable(function(agent) { /* ... */ });
 ```
 Subscribe a method to be called when the agent becomes not-routable, meaning
 that they are online but cannot be routed incoming contacts.
 
 ### `agent.onOffline()`
-```
-agent.onOffline(function(agent) { ... });
+```js
+agent.onOffline(function(agent) { /* ... */ });
 ```
 Subscribe a method to be called when the agent goes offline.
 
 ### `agent.onError()`
+```js
+agent.onError(function(agent) { /* ... */ });
 ```
-agent.onError(function(agent) { ... });
-```
-Subscribe a method to be called when the agent is put into an error state.  This
+Subscribe a method to be called when the agent is put into an error state. This
 can occur if Streams is unable to get new agent data, or if the agent fails to
-accept an incoming contact, or in other error cases.  It means that the agent is
+accept an incoming contact, or in other error cases. It means that the agent is
 not routable, and may require that the agent switch to a routable state before
 being able to be routed contacts again.
 
 ### `agent.onSoftphoneError()`
-```
-agent.onSoftphoneError(function(agent) { ... });
+```js
+agent.onSoftphoneError(function(error) { /* ... */ });
 ```
 Subscribe a method to be called when the agent is put into an error state specific to softphone funcionality.
 
+The `error` argument is a `connect.SoftphoneError` instance with the following methods: `getErrorType()`, `getErrorMessage()`, `getEndPointUrl()`.
+
 ### `agent.onAfterCallWork()`
+```js
+agent.onAfterCallWork(function(agent) { /* ... */ });
 ```
-agent.onAfterCallWork(function(agent) { ... });
-```
-Subscribe a method to be called when the agent enters the "After Call Work" (ACW) state.  This is a non-routable state which exists to allow agents some time to wrap up after handling a contact before they are routed additional contacts.
+Subscribe a method to be called when the agent enters the "After Call Work" (ACW) state. This is a non-routable state which exists to allow agents some time to wrap up after handling a contact before they are routed additional contacts.
 
 ### `agent.getState()` / `agent.getStatus()`
-```
-var state = agent.getState()
+```js
+var state = agent.getState();
 ```
 Get the agent's current `AgentState` object indicating their availability state type.
 This object contains the following fields:
 
+* `agentStateARN`: The agent's current state ARN.
 * `name`: The name of the agent's current availability state.
+* `startTimestamp`: A `Date` object that indicates when the state was set.
 * `type`: The agent's current availability state type, as per the `AgentStateType` enumeration.
-* `duration`: A relative local state duration.  To get the actual duration of the state relative
-  to the current time, use `agent.getStateDuration()`.
```

Comment:
Found out that the `duration` field was replaced by `startTimestamp` (or it was originally a typo).